### PR TITLE
stream: added Interleaver, Deinterleaver

### PR DESCRIPTION
**Pull request by ncorgan**
* Closes https://github.com/pothosware/PothosBlocks/issues/38
* Closes https://github.com/pothosware/PothosComms/issues/23

### CODE REVIEW

File: stream/Interleaver.cpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                return inputPort->buffer().convert(_outputDType);
+            });
+
+        auto minElemIter = std::min_element(
+                               convertedInputs.begin(),
+                               convertedInputs.end(),
+                               [](const Pothos::BufferChunk& chunk1, const Pothos::BufferChunk& chunk2)
+                               {
+                                   return (chunk1.elements() < chunk2.elements());
+                               });
```
Comment (guruofquality) [inline stream/Interleaver.cpp]: Have you seen workinfo? https://github.com/pothosware/PothosCore/blob/master/include/Pothos/Framework/Block.hpp#L215

It already has a vector of input pointers and min/max element calculations: https://github.com/pothosware/PothosCore/blob/master/include/Pothos/Framework/WorkInfo.hpp#L23

**Comment by ncorgan** [comment]
Do those numbers come from the InputPort or underlying BufferChunk? The InputPort’s type is unspecified, so its element count is in bytes. It would need to check the underlying BufferChunk, which is whatever type the previous block passed in.

**Comment by guruofquality** [comment]
Yea, the workInfo elements are based on the port elements counts, so its all bytes in this case. Only useful if the input types are homogeneous. The work info will also give you an array pointers, which I guess in this case, you can use.

**Comment by ncorgan** [comment]
Taking another look at these, I still need the BufferChunks for Interleaver so I can call convert, and for Deinterleaver, the code comes out to about the same, since I need to convert all the output void* pointers to uint8_t* for pointer arithmetic.

**Comment by guruofquality** [comment]
I didnt catch that, makes sense

**Pull request merged**